src/main/java/com/softvision/books/services/domain/Pagination.java

```diff
@@ -0,0 +1,32 @@
+package com.softvision.books.services.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Pagination<T> {
```

Comment:
May not need to implement Pagination class as the JpaRepository and Pageable object should be able to support paging and sorting (passing or not the pageable params should work), something like below:

@Repository
public interface BookRepository extends JpaRepository<Book, Long> {
    Page<Book> findByAuthorId(Long authorId, Pageable pageable);
    Page<Book> findByIdAndAuthorId(Long id, Long authorId, Pageable pageable);